android/app/src/main/java/com/google/samples/cronet_sample/MainActivity.java

```diff
@@ -27,47 +27,62 @@
 import androidx.recyclerview.widget.RecyclerView;
 import androidx.swiperefreshlayout.widget.SwipeRefreshLayout;
 import com.google.samples.cronet_sample.data.ImageRepository;
+import java.io.File;
+import java.io.IOException;
 import java.util.concurrent.atomic.AtomicLong;
+import org.chromium.net.CronetEngine;
 
 public class MainActivity extends AppCompatActivity {
 
     private static final String TAG = "MainActivity";
     private SwipeRefreshLayout swipeRefreshLayout;
-    private AtomicLong cronetLatency = new AtomicLong();
+    private final AtomicLong cronetLatency = new AtomicLong();
     private long totalLatency;
     private long numberOfImages;
-    private ViewAdapter viewAdapter;
+    public static CronetEngine cronetEngine;
```

Comment:
Hmm, I'm not really convinced the engine instance should be static given it's tied to a lifespan of a single instance. Why have you made it so?

Reply:
I had my reservations about doing this, but then I opined that only one instance of MainActvity is created for each device. Thinking about it now, I am wrong because an instance can be destroyed with onDestroy() and another created with onCreate(). I will change it.